### ISSUE TITLE
Nullable static return typehint causes whitespace error

### ISSUE THREAD

**Issue by jnoordsij** [squizlabs/PHP_CodeSniffer#3277]
**Describe the bug**
Since PHP8, static is allowed as return type. In https://github.com/squizlabs/PHP_CodeSniffer/pull/3189 a fix was added to prevent this from yielding whitespace errors. However, a nullable static return type still produces an error.

**Code sample**
```php
<?php

namespace App\Example;

class Example
{
    # works fine now
    public function testA(): static
    {
        return new static();
    }

    # Scope keyword "static" must be followed by a single space; found newline
    public function testB(): ?static
    {
        return new static();
    }
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
    <rule ref="PSR12"/>
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php`
3. See error message displayed
```
FOUND 1 ERROR AFFECTING 1 LINE
--------------------------------------------------------------------------------

 14 | ERROR | [x] Scope keyword "static" must be followed by a single space;
    |       |     found newline
--------------------------------------------------------------------------------
```

**Expected behavior**
No error

**Versions (please complete the following information):**
 - OS: Windows 10 & Alpine 3.13
 - PHP: 8.0
 - PHPCS: master
 - Standard: PSR12

**Comment by gsherwood** [comment]
Thanks for picking this up. I've committed a fix and it will be in 3.6.0.

**Comment by jnoordsij** [comment]
Thanks for the quick fix! Looking forward to the release.
